### PR TITLE
Kth largest

### LINKED ISSUE #4890: Kth largest element in an array
## 💥 Proposal
I'll solve this problem using heap in language c++.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

### PR DESCRIPTION
**Pull request by shonali2600**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

I've added code to find Kth Largest Element in an array using heap.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

closes #4890

### CODE REVIEW

File: C-Plus-Plus/heap/KthLargestElement.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/*
+  Code to find the Kth Largest element in an array is given below.
+*/
+
+#include <bits/stdc++.h>
+#include <queue>
```
Comment (RaiyanMahin) [inline C-Plus-Plus/heap/KthLargestElement.cpp]: remove `#include <queue>` and `#include <vector>`

File: C-Plus-Plus/heap/KthLargestElement.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#include <vector>
+
+using namespace std;
+int findKthLargest(vector<int> &a, int k)
```
Comment (RaiyanMahin) [inline C-Plus-Plus/heap/KthLargestElement.cpp:13]: add a new line and a bit description about the function

File: C-Plus-Plus/heap/KthLargestElement.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    //creating a min heap
+    priority_queue<int, vector<int>, greater<int>> heap;
+
+    for (auto u : a) //iterating through vector
+    {
+        if (heap.size() < k)
+            heap.push(u); //keep filling the heap till K
+        else
+        {
+            if (u > heap.top()) //checking if there is another larger element that top of the heap remaining in the vector array.
```
Comment (RaiyanMahin) [inline C-Plus-Plus/heap/KthLargestElement.cpp]: use multi line comment

File: C-Plus-Plus/heap/KthLargestElement.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            {
+                heap.pop();
+                heap.push(u);
+            }
+        }
+    }
+    return heap.top(); //The element on the top of the heap after traversing the whole array will be the Kth Largest ELement.
```
Comment (RaiyanMahin) [inline C-Plus-Plus/heap/KthLargestElement.cpp]: use multi line comment

File: C-Plus-Plus/heap/KthLargestElement.cpp [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+int main()
+{
+    cout << "Enter the size of the vector array: ";
```
Comment (RaiyanMahin) [inline C-Plus-Plus/heap/KthLargestElement.cpp:37]: use proper variable names

File: C-Plus-Plus/heap/KthLargestElement.cpp [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+*/
+
+#include <bits/stdc++.h>
+#include <queue>
+#include <vector>
+
+using namespace std;
+int findKthLargest(vector<int> &a, int k)
+{
+
```
Comment (RaiyanMahin) [inline C-Plus-Plus/heap/KthLargestElement.cpp]: remove extra lines of code

**Review by RaiyanMahin — CHANGES_REQUESTED** [review]

**Comment by shonali2600** [comment]
I've done all the required changes. Please review my PR again.

File: C-Plus-Plus/heap/KthLargestElement.cpp [hunk 7]
```diff
@@ -0,0 +1,67 @@
+/*
+  Code to find the Kth Largest element in an array is given below.
+*/
+
+#include <bits/stdc++.h>
+
+using namespace std;
+
+/*
```
Comment (RaiyanMahin) [inline C-Plus-Plus/heap/KthLargestElement.cpp:9]: only one line description of your function.

File: C-Plus-Plus/heap/KthLargestElement.cpp [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+{
+    priority_queue<int, vector<int>, greater<int>> heap;       //creating a min heap
+
+    for (auto u : a) //iterating through vector
+    {
+        if (heap.size() < k)
+            heap.push(u); //keep filling the heap till K
+        else
+        {
+            if (u > heap.top()) /*checking if there is another element larger
```
Comment (RaiyanMahin) [inline C-Plus-Plus/heap/KthLargestElement.cpp:27]: use multi line comment.

File: C-Plus-Plus/heap/KthLargestElement.cpp [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+                                  than top of the heap remaining in the vector array. */
+            {
+                heap.pop();
+                heap.push(u);
+            }
+        }
+    }
+    return heap.top(); /*The element on the top of the heap after
```
Comment (RaiyanMahin) [inline C-Plus-Plus/heap/KthLargestElement.cpp:35]: use multi line comment.

**Review by RaiyanMahin — CHANGES_REQUESTED** [review]

**Comment by shonali2600** [comment]
Done, now please review my PR.

**Review by RaiyanMahin — APPROVED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Problem already available in Cpp section. Kindly check before opening any issue.

**Comment by shonali2600** [comment]
@ankitaggarwal23 , No it is not there, I've checked. I've created pull request for "Kth largest" element, and in CPP section "Kth smallest" element program is present.

**Comment by ankitaggarwal23** [comment]
Kindly check it properly it there I checked and then only I'm telling [here](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/other/Kth_largest_element.cpp)

**Comment by RaiyanMahin** [comment]
Ankit told about this one :
https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/other/Kth_largest_element.cpp

**Comment by shonali2600** [comment]
ohh okay, i got it, I'm so sorry I was checking inside the heap folder and it is present inside the other folder

**Comment by HarshCasper** [comment]
Hi @shonali2600 

What about writing an article on it? You can create an issue on NeoAlgo-Docs and get started with that :wink:

**Comment by shonali2600** [comment]
@HarshCasper Thanks for giving me the suggestion, I'll create an issue on that for the article.

**Pull request closed without merging**